torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -472,7 +472,14 @@ auto FunctionParameter::check(PyObject* obj, std::vector<py::handle> &overloaded
     }
     case ParameterType::INT_LIST: {
       if (PyTuple_Check(obj) || PyList_Check(obj)) {
-        return true;
+        if (PySequence_Size(obj) == 0) {
+          return true;
+        }
+        auto item = py::reinterpret_steal<py::object>(
+            PySequence_GetItem(obj, 0));
+        // NOTE: JIT tracer allows arbitrary scalar tensors to act as ints
+        // in an intlist argument Even float or complex scalars
+        return (THPVariable_Check(item.ptr()) || THPUtils_checkIndex(item.ptr()));
```

Comment:
Be good to factor this out now that it's nontrivial, a la `is_float_or_complex_list()` just below